### DISCUSSION TITLE
[Postgres] I want array of ids

### DISCUSSION THREAD

**Discussion by SupertigerDev** [prisma/prisma#15118] [Q&A]
So, this is my model:

```prisma
model Member {
  id       String @id
  username String

  roleIds String[]
  roles   Role[]      @relation(fields: [roleIds], references: [id])
}

model Role {
  id   String @id
  name String

  memberIds String[]
  members   Member[] @relation(fields: [memberIds], references: [id])
}
```

I get this error: `Error validating: Embedded many-to-many relations are not supported on Postgres. Please use the syntax defined in https://pris.ly/d/relational-database-many-to-many`

I followed the link, From what I can see, it doesn't show me how I could still get an array of Ids (memberIds or roleIds).

Is there anyway?

**Comment by notscar** [comment]
What you're trying to do is impossible, and retriving userIds and roleIds should be done via fetching the data with the `select` field

```prisma
model Member {
  id       String @id
  username String

  roles Role[] 
}

model Role {
  id   String @id
  name String

  members   Member[]
}
```

```js
await prisma.member.findUnique({
    where: {
        id: "myid"
    },
    select: {
        username: true,
        roles: {
            select: {
                id: true
            }
        }
    }
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
